pulp_rpm/app/models/package.py

```diff
@@ -310,7 +310,22 @@ def createrepo_to_dict(cls, package):
             changelog_limit = settings.KEEP_CHANGELOG_LIMIT or 1
             # changelogs are listed in chronological order, grab the last N changelogs from the list
             changelogs = changelogs[-changelog_limit:]
-
+        files = getattr(package, CR_PACKAGE_ATTRS.FILES, [])
+        seen = []
+        for path in files:
+            if path in seen:
```

Comment:
This will be really expensive because these lists can get pretty long.  It would be better to use a set as before - we just don't want to use that set as the collection itself.

It's also important to note that it's not just a list of strings, it's a list of tuples, so `path` isn't an accurate name.

Something like

```Python
has_duplicates = False
seen = set()
deduplicated_files = []
for fileentry in files:
    if fileentry in seen:
        has_duplicates = True
    else:
        deduplicated_files.append(fileentry)
files = deduplicated_files
```

Reply:
the list will have the same size as the set, as we only append if it isn't duplicated

Reply:
Right, but the lookup itself is O(N) for the list vs. O(1) for the set.

Reply:
I didn't know about that, where can I learn about the big O of python internals?

Reply:
It's usually pretty applicable across languages because the implementations are generally similar, but it looks like there's a Python-specific table here: https://wiki.python.org/moin/TimeComplexity

Reply:
thank you!